### PR TITLE
Bugfix for verbatim-identifiers and contextual keywords, always-false case "value.IsMissing" removed

### PR DESCRIPTION
**Pull request by bernd5**
Fixes handling for identifiers like "_", "@class" and "nameof"

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
Nice! Care to add a couple of unit-tests? It should be pretty easy.

**Comment by bernd5** [comment]
It is not really easy to write a unit test. The problem is that "ToString()" worked correctly - but the processing in the semantic model expects that the ContextualKeyword-Kind is set and the "ValueText" has no "@"...

File: src/Quoter/Quoter.cs [hunk 1]
```diff
@@ -385,6 +385,15 @@ private ApiCall QuoteList(IEnumerable syntaxList, string name)
             return codeBlock;
         }
 
+        private static SyntaxKind GetContextualKind(SyntaxToken value)
+        {
+            var property = typeof(SyntaxToken).GetProperty("RawContextualKind",
+                BindingFlags.NonPublic | BindingFlags.Instance);
+            var result = property.GetValue(value);
+
+            return (SyntaxKind)(int)result;
```
Comment (CyrusNajmabadi) [inline src/Quoter/Quoter.cs:394]: Can you just use GetContextualKeywordKind with the text of the token instead?

Comment (bernd5) [reply src/Quoter/Quoter.cs]: Thanks - I updated the implementation...

File: src/Quoter/Quoter.cs [hunk 2]
```diff
@@ -422,26 +413,35 @@ private ApiCall QuoteToken(SyntaxToken value, string name)
...
                     trailing = trailing ?? GetEmptyTrivia("TrailingTrivia");
 
                     arguments.Add(leading);
-                    arguments.Add(contextualKind);
-                    arguments.Add(actualValue);
+                    arguments.Add(kind);
+                    arguments.Add(escapedTokenText);
                     arguments.Add(EscapeAndQuote(value.ValueText));
                     arguments.Add(trailing);
+                } else
```
Comment (KirillOsenkov) [inline src/Quoter/Quoter.cs:428]: mind formatting the file? it should place else on a separate line

**Comment by bernd5** [comment]
Anything to add or change?

**Review by KirillOsenkov — APPROVED** [review]

**Comment by KirillOsenkov** [comment]
Thanks!

**Pull request merged**